internal/provider/resource_tfe_workspace_policy_set_exclusion.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceTFEWorkspacePolicySetExclusion() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEWorkspacePolicySetExclusionCreate,
+		Read:   resourceTFEWorkspacePolicySetExclusionRead,
+		Delete: resourceTFEWorkspacePolicySetExclusionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceTFEWorkspacePolicySetExclusionImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"policy_set_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"workspace_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceTFEWorkspacePolicySetExclusionCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+
+	policySetID := d.Get("policy_set_id").(string)
+	workspaceExclusionID := d.Get("workspace_id").(string)
+
+	policySetAddWorkspaceExclusionsOptions := tfe.PolicySetAddWorkspaceExclusionsOptions{}
+	policySetAddWorkspaceExclusionsOptions.WorkspaceExclusions = append(policySetAddWorkspaceExclusionsOptions.WorkspaceExclusions, &tfe.Workspace{ID: workspaceExclusionID})
+
+	err := config.Client.PolicySets.AddWorkspaceExclusions(ctx, policySetID, policySetAddWorkspaceExclusionsOptions)
+	if err != nil {
+		return fmt.Errorf(
+			"error attaching policy set id %s to excluded workspace %s: %w", policySetID, workspaceExclusionID, err)
```

Comment:
I believe this error should be scoped to the behavior we are expecting - Add workspace exclusions to a policy set. Something like:
```
error adding workspace exclusion to policy set
```